### DISCUSSION TITLE
fixed and linked, apps and packages dependencies

### DISCUSSION THREAD

**Discussion by Karnak19** [changesets/changesets#1072] [Q&A]
Hello there ! 😃 

I am currently doing research, planning to migrate my multiple replicated projects under a monorepo (with Turborepo).

My file tree will looks like this:
```
├── apps
│   ├── project-A
│   ├── project-B
│   └── project-C
├── packages
│   ├── api
│   └── ui
```

My needs are the following:
- When I update `ui`, both projects ABC depends on it, so they should have their version bumped too.
- When I update let's say `project-A`, only this one should have his version bumped. not the 2 others projects, nor the `ui` package.

This is where I think is the limit with changesets ? What I would like to write in the config would be:
```json
{
  "linked": [
    ["ui", "project-A"],
    ["ui", "project-B"],
    ["ui", "project-C"]
  ],
}
```
But I got an error from the CLI when doing so

Is this something possible with changesets, or should I look for another solution ?

Thanks in advance 🙏

**Comment by Andarist** [comment]
With default settings, the versions are bumped if dependency ranges stop being satisfied, see the options [here](https://github.com/changesets/changesets/blob/e4204820cdb93d22a3c7a27f452a0ab30db6369a/docs/experimental-options.md#updateinternaldependents-type-out-of-range--always). And this is graph-based - so if pkg A depends on pkg B and pkg B gets a new major version then we compute that we also have to create a patch bump for A, this can happen recursively, etc.

`linked` option determines the collective version for a whole group of packages and when something **has to** be published it uses that collective version instead of the min one. So a package within a group that was not published lately could stay on, let's say 1.2.0, while other packages in the group could have been bumped on multiple occasions in the meantime and they could got to, let's say, 1.7.0. So now when this package gets a minor bump - it will be bumped to 1.8.0 (instead of 1.3.0).

On the other hand, `fixed` just grabs the whole group and bumps all packages in the group with the current highest bump. Even if a package doesn't have to be published (ranges are satisfied etc) - it will still be published. This is also different from `updateInternalDependents: 'always'` - because this doesn't even require packages in the group to depend on each other.

**Comment by Karnak19** [comment]
So if I understand correctly, by making my config file like this
```json
{
  "$schema": "https://unpkg.com/@changesets/config@2.3.0/schema.json",
  "changelog": "@changesets/cli/changelog",
  "commit": false,
  "fixed": [],
  "linked": [],
  "access": "restricted",
  "baseBranch": "main",
  "updateInternalDependencies": "patch",
  "___experimentalUnsafeOptions_WILL_CHANGE_IN_PATCH": {
    "updateInternalDependendents": "always"
  },
  "ignore": []
}
```

should fit my needs, but maybe I miss something, when i make an update on my `ui` package, it only bump the `ui` package, not my `project-A` nor `project-B` that both have `ui` as dependency 🤔 

[Here is a repo](https://github.com/netcosports/ott-turbo) to demonstrate, everything is not really clear to me 🥲
